libs/SalesforceSDK/src/com/salesforce/androidsdk/security/PasscodeManager.java

```diff
@@ -616,12 +616,14 @@ public void setMinPasscodeLength(Context ctx, int minPasscodeLength) {
      */
     public void setPasscodeLength(Context ctx, int passcodeLength) {
     	if (passcodeLength > this.passcodeLength) {
-            if (hasStoredPasscode(ctx)) {
+            if (hasStoredPasscode(ctx) && passcodeLengthKnown) {
                 this.passcodeChangeRequired = true;
             }
+
+            this.passcodeLength = passcodeLength;
     	}
-        this.passcodeLength = passcodeLength;
```

Comment:
@bhariharan Why was passcode length requirement allowed to be lowered?

Reply:
Passcode length could be lowered if you log out of an account with a longer length requirement, when there are 2 accounts with passcode enabled.